src/Route.php

```diff
@@ -23,24 +23,26 @@ final class Route
     /**
      * @var callable[]|string[]|array[]
      */
-    private array $middlewares = [];
+    private array $middlewareDefinitions = [];
     private array $defaults = [];
 
     private function __construct(?DispatcherInterface $dispatcher = null)
     {
         $this->dispatcher = $dispatcher;
     }
 
-    public function withDispatcher(DispatcherInterface $dispatcher): self
+    public function injectDispatcher(DispatcherInterface $dispatcher): void
```

Comment:
Can it be immutable?

Reply:
> Can it be immutable?

No, it doesn't. We should reuse the stack on the next request in the request loop.